src/graphics/shader.cpp

```diff
@@ -0,0 +1,119 @@
+//
+// Created by Lukasz Gut on 22/12/2018.
+//
+
+#include "shader.h"
+#include <glad/glad.h>
+#include "pch.h"
+#include "file_manager.h"
+
+namespace Bald::Graphics {
+
+    Shader::Shader(const char *vertexPath, const char *fragmentPath)
+    : m_ShaderID(0), m_VertexPath(vertexPath), m_FragmentPath(fragmentPath) {
+        m_ShaderID = CreateShader();
+    }
+
+    Shader::~Shader() {
+        glDeleteProgram(m_ShaderID);
+    }
+
+    void Shader::Bind() const {
+        glUseProgram(m_ShaderID);
+    }
+
+    void Shader::Unbind() const {
+        glUseProgram(0);
+    }
+
+    void Shader::SetUniform4f(const char* uniformName, float v0, float v1, float v2, float v3) const {
+        glUniform4f(GetUniformLocation(uniformName), v0, v1, v2, v3);
+    }
+
+    inline unsigned int Shader::GetShaderID() const {
+        return m_ShaderID;
+    }
+
+    unsigned int Shader::CreateShader() const {
+        using namespace Bald;
```

Comment:
that's unnecessary because you already are in this namespace

Reply:
LMAO, fixed :D